runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingModeExecutionContext.java

```diff
@@ -357,7 +358,7 @@ public void invalidateCache() {
         }
       }
       activeReader = null;
-      stateCache.invalidate(key);
+      stateCache.invalidate(key, getWork().getShardingKey());
```

Comment:
the cachedReader is also keyed: https://github.com/apache/beam/blob/88acc5267f759d81e9836a9db17b9e0ee521c785/runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingModeExecutionContext.java#L338-L340
Should it also be keyed by sharedKey?

Reply:
Good catch! Fixed.